filters/builtin/mod_header.go

```diff
@@ -73,3 +73,60 @@ func (f *modRequestHeader) Request(ctx filters.FilterContext) {
 }
 
 func (*modRequestHeader) Response(filters.FilterContext) {}
+
+type modResponseHeader struct {
+	headerName  string
+	rx          *regexp.Regexp
+	replacement string
+}
+
+// NewModResponseHeader returns a new filter Spec, whose instances execute
+// regexp.ReplaceAllString on the request host. Instances expect three
+// parameters: the header name, the expression to match and the replacement string.
+// Name: "modResponseHeader".
+func NewModResponseHeader() filters.Spec { return &modResponseHeader{} }
+
+func (spec *modResponseHeader) Name() string {
+	return filters.ModResponseHeaderName
+}
+
+//lint:ignore ST1016 "spec" makes sense here and we reuse the type for the filter
+func (spec *modResponseHeader) CreateFilter(config []interface{}) (filters.Filter, error) {
+	if len(config) != 3 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	headerName, ok := config[0].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	expr, ok := config[1].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	replacement, ok := config[2].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	rx, err := regexp.Compile(expr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &modResponseHeader{headerName: headerName, rx: rx, replacement: replacement}, nil
+}
+
+func (*modResponseHeader) Request(filters.FilterContext) {}
+
+func (f *modResponseHeader) Response(ctx filters.FilterContext) {
+	resp := ctx.Response()
+
+	if _, ok := resp.Header[http.CanonicalHeaderKey(f.headerName)]; !ok {
+		return
+	}
```

Comment:
There is a highlevel https://pkg.go.dev/net/http#Header.Get

Reply:
Yes, there is but it returns empty string in case of key is not present in header. I think it is better to leave header unpresent in this case than implicitly create a header with empty string value.

Reply:
I think the edge case in case there is a header with empty string as value and you do a modification like:
```
/.*/foo/
```
If there is no header it should stay no header and if the header is present with empty string it should set the value to "foo".